File: ejercicio/completar.py
"""
🐍 Ejercicio de Python - Prueba de Familiarización
===================================================

Tu tarea es completar la función saludar() que aparece abajo.

La función debe:
- Recibir un parámetro 'nombre' (string)
- Retornar un saludo con el formato: "¡Hola, {nombre}! Bienvenido/a a la prueba."

Ejemplo:
- saludar("María") → "¡Hola, María! Bienvenido/a a la prueba."
- saludar("Juan")  → "¡Hola, Juan! Bienvenido/a a la prueba."
"""


def saludar(nombre):
    """
    Retorna un saludo personalizado para el nombre dado.
    
    Args:
        nombre (str): El nombre de la persona a saludar
        
    Returns:
        str: Un saludo con el formato "¡Hola, {nombre}! Bienvenido/a a la prueba."
    """
    # TODO: Implementa esta función
    # Reemplaza 'pass' con tu código
    # Hint: Usa f-strings para formatear el string de salida
    pass


# ============================================
# Código de verificación - NO MODIFICAR
# Ejecuta: python ejercicio/completar.py
# ============================================
if __name__ == "__main__":
    print("🧪 Probando la función saludar()...\n")
    
    # Test 1
    resultado1 = saludar("María")
    esperado1 = "¡Hola, María! Bienvenido/a a la prueba."
    if resultado1 == esperado1:
        print("✅ Test 1 PASÓ: saludar('María')")
    else:
        print(f"❌ Test 1 FALLÓ: saludar('María')")
        print(f"   Esperado: {esperado1}")
        print(f"   Obtenido: {resultado1}")
    
    # Test 2
    resultado2 = saludar("Juan")
    esperado2 = "¡Hola, Juan! Bienvenido/a a la prueba."
    if resultado2 == esperado2:
        print("✅ Test 2 PASÓ: saludar('Juan')")
    else:
        print(f"❌ Test 2 FALLÓ: saludar('Juan')")
        print(f"   Esperado: {esperado2}")
        print(f"   Obtenido: {resultado2}")
    
    print("\n" + "="*50)
    if resultado1 == esperado1 and resultado2 == esperado2:
        print("🎉 ¡Todos los tests pasaron! Tu función está correcta.")
    else:
        print("💡 Revisa tu implementación y vuelve a intentar.")

File: ejercicio/completar.js
/**
 * 📦 Ejercicio de JavaScript - Prueba de Familiarización
 * =======================================================
 * 
 * Tu tarea es completar las funciones sumar() y esPar() que aparecen abajo.
 * 
 * Función sumar(a, b):
 * - Debe retornar la suma de a + b
 * - Ejemplo: sumar(2, 3) → 5
 * 
 * Función esPar(numero):
 * - Debe retornar true si el número es par
 * - Debe retornar false si el número es impar
 * - Ejemplo: esPar(4) → true
 * - Ejemplo: esPar(7) → false
 */


/**
 * Retorna la suma de dos números.
 * @param {number} a - Primer número
 * @param {number} b - Segundo número
 * @returns {number} La suma de a + b
 */
function sumar(a, b) {
    // TODO: Implementa esta función
    // Reemplaza 'return null' con tu código
    return null;
}


/**
 * Determina si un número es par.
 * @param {number} numero - El número a verificar
 * @returns {boolean} true si es par, false si es impar
 */
function esPar(numero) {
    // TODO: Implementa esta función
    // Reemplaza 'return null' con tu código
    // Hint: Usa el operador módulo (%) para verificar si es divisible por 2
    return null;
}


// Exportar funciones para los tests
module.exports = { sumar, esPar };


// ============================================
// Código de verificación - NO MODIFICAR
// Ejecuta: node ejercicio/completar.js
// ============================================
if (require.main === module) {
    console.log("🧪 Probando las funciones...\n");
    
    let todosLosTestsPasaron = true;
    
    // Tests para sumar()
    console.log("--- Función sumar() ---");
    
    const resultado1 = sumar(2, 3);
    if (resultado1 === 5) {
        console.log("✅ Test 1 PASÓ: sumar(2, 3) = 5");
    } else {
        console.log(`❌ Test 1 FALLÓ: sumar(2, 3)`);
        console.log(`   Esperado: 5`);
        console.log(`   Obtenido: ${resultado1}`);
        todosLosTestsPasaron = false;
    }
    
    const resultado2 = sumar(-1, 1);
    if (resultado2 === 0) {
        console.log("✅ Test 2 PASÓ: sumar(-1, 1) = 0");
    } else {
        console.log(`❌ Test 2 FALLÓ: sumar(-1, 1)`);
        console.log(`   Esperado: 0`);
        console.log(`   Obtenido: ${resultado2}`);
        todosLosTestsPasaron = false;
    }
    
    // Tests para esPar()
    console.log("\n--- Función esPar() ---");
    
    const resultado3 = esPar(4);
    if (resultado3 === true) {
        console.log("✅ Test 3 PASÓ: esPar(4) = true");
    } else {
        console.log(`❌ Test 3 FALLÓ: esPar(4)`);
        console.log(`   Esperado: true`);
        console.log(`   Obtenido: ${resultado3}`);
        todosLosTestsPasaron = false;
    }
    
    const resultado4 = esPar(7);
    if (resultado4 === false) {
        console.log("✅ Test 4 PASÓ: esPar(7) = false");
    } else {
        console.log(`❌ Test 4 FALLÓ: esPar(7)`);
        console.log(`   Esperado: false`);
        console.log(`   Obtenido: ${resultado4}`);
        todosLosTestsPasaron = false;
    }
    
    console.log("\n" + "=".repeat(50));
    if (todosLosTestsPasaron) {
        console.log("🎉 ¡Todos los tests pasaron! Tus funciones están correctas.");
    } else {
        console.log("💡 Revisa tu implementación y vuelve a intentar.");
    }
}
